packages/core/core/src/services/content-api/index.ts

```diff
@@ -70,6 +70,9 @@ const createContentAPI = (strapi: Core.Strapi) => {
   };
 
   const sanitizer = sanitize.createAPISanitizers({
+    getModel(uid: string) {
```

Comment:
3 problems:

- the utils are generic and should not depend on UID.Schema as you could pass a map of other things (that's why this traversal should be with the content-types domain and not so generic) 
- UID.Schema doesn't extend string apparently so I can't expect a string in the utils and send a UID.Schema (make you can fix it that would be very convenient when using generic code that shouldn't know anything about UID.Schema in the 1st place anyway.
- The utils can't depend on @strapi/types for now as it creates a circular dep so I can't import UID.Schema to make the types identical. And I would rather not compy the UID.Schema definition

Reply:
>the utils are generic and should not depend on UID.Schema as you could pass a map of other things (that's why this traversal should be with the content-types domain and not so generic)

Not sure what other kind of model we could expect to traverse (+ the parameter name is UID), but fair enough

> UID.Schema doesn't extend string apparently so I can't expect a string in the utils and send a UID.Schema (make you can fix it that would be very convenient when using generic code that shouldn't know anything about UID.Schema in the 1st place anyway.

Well, that's made on purpose to avoid sending malformatted UIDs to the utils expecting one.
Instead of having looser types, I would rather have type guards asserting that a string is a valid UID.

As for the other direction, you can still send UID.Schema to a function expecting a string
![image](https://github.com/strapi/strapi/assets/25851739/4dcdf6c3-2f37-4a2d-ab76-32824c27a309)

> The utils can't depend on @strapi/types for now as it creates a circular dep so I can't import UID.Schema to make the types identical. And I would rather not compy the UID.Schema definition

Alright

Reply:
Nope I can't send a UID.Schema to the function expecting a string or I wouldn't have a problem in the code right now